modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -2476,35 +2476,44 @@ def drop(self, index=None, columns=None, errors: str = "raise"):
 
     def duplicated(self, **kwargs):
         def _compute_hash(df):
-            return df.apply(
+            result = df.apply(
                 lambda s: hashlib.new("md5", str(tuple(s)).encode()).hexdigest(), axis=1
-            ).to_frame()
+            )
+            if isinstance(result, pandas.Series):
+                result = result.to_frame(
+                    result.name
+                    if result.name is not None
+                    else MODIN_UNNAMED_SERIES_LABEL
+                )
```

Comment:
This block of code is used in two places, let's reuse it somehow.

Reply:
This code snippet is used in many places in the qc. If we want it to be reused, let's create a separate issue for that.

Reply:
If used in many places, then ok, let's create

Reply:
https://github.com/modin-project/modin/issues/5741